doc/specification/language-proposal.md

```diff
@@ -0,0 +1,1518 @@
+
+**Function scope.**
+Function arguments have function scope, which extends until the end of the function definition.
+
+**Regime scope.**
+All top-level regime definitions in an interface have as scope the entire interface definition, even preceding the regime definition. The same applies within a region definition: the scope of an internal region identifier is the entire body of the region definition. An internal region identifier is visible as a _qualified-identifier_ elsewhere within the same interface definition.
+
+**Expression scope.**
+`let` and `with` binding expressions introduce new identifiers that have as scope the final expression in the binding.
+
+**Record scope.**
+Field names introduced in a record type or record value have record scope, which extends to the end of the record type or record value construction.
+
+Scope form a hierarchy of inner and outer scopes, with the outermost scope being the global scope.
+
+### Context
+<a id="context"/>
+
+Two different identifiers with the same ***symbol*** may be bound in the same scope, if they are in different _contexts_. There are five distinct contexts in Arblang:
+
+1. _Module_ context applies wherever a module name is being introduced in a module definition, or where a module is referenced in a module import clause.
+2. _Type context_ applies in type expressions and where a type alias is introduced.
+3. _Regime context_ applies in regime clauses and where a regime name is introduced in a regime definition.
+4. _Record context_ applies to field names within a record type description or in a record value construction.
+5. _Expression context_ applies within value expressions, function argument definitions, the definition of parameters, constants, functions, and in the association of an identifier with an imported module.
+
+Each context can be regarded as constituting a different namespace for identifiers.
+
+### Masking
+<a id="masking"/>
+
+An identifier bound in a given context in an outer scope may be _masked_ by a binding of an identifier with the same ***symbol*** in an inner scope. This is permitted only in the following circumstances:
+
+1. An identifier bound in expression context in a function scope by a function argument or in expression scope by a `let` or `with` binding may mask an identifier in an outer scope.
+2. An identifier bound in regime context in an inner regime definition may mask a regime identifier in an outer regime or interface scope.
+
+It is otherwise an error to bind an identifier with the same ***symbol*** as an already bound identifier in the same context.
```

Comment:
Question
```
module A { type T; }
module B { type T; }
module C {
  import A;
  import B;
}
```
should fail. However, 
```
module C {
  import A as A;
  import B;
}
```
should be OK. Correct?

Reply:
As proposed, `import A` is exactly equivalent to `import A as A`.